sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaSource.java

```diff
@@ -293,174 +287,44 @@ public boolean start() throws IOException {
-          String key = metadataIterator.next();
-          // The metadata name/value separator can be configured if needed
-          current = key + "=" + tikaMetadata.get(key);
-      }
-      return current != null;
+      return false;
     }
 
     @Override
-    public String getCurrent() throws NoSuchElementException {
+    public ParseResult getCurrent() throws NoSuchElementException {
       if (current == null) {
         throw new NoSuchElementException();
       }
-      return current;
+      return new ParseResult(filePath, current, tikaMetadata);
     }
 
     @Override
-    public void close() throws IOException {
-      if (execService != null) {
-          execService.shutdown();
-      }
-    }
-
-    ExecutorService getExecutorService() {
-      return execService;
+    public BoundedSource<ParseResult> getCurrentSource() {
+      return source;
     }
 
     @Override
-    public BoundedSource<String> getCurrentSource() {
-      return source;
+    public void close() throws IOException {
+      // complete
     }
   }
 
   /**
    * Tika Parser Content Handler.
    */
-  static class ContentHandlerImpl extends DefaultHandler {
-    private Queue<String> queue = new ConcurrentLinkedQueue<>();
-    private volatile boolean documentEnded;
-    private volatile Exception parseException;
-    private volatile String current;
-    private int minTextLength;
-
-    @Override
-    public void characters(char ch[], int start, int length) throws SAXException {
-      String value = new String(ch, start, length).trim();
-      if (!value.isEmpty()) {
-        if (minTextLength <= 0) {
-          queue.add(value);
-        } else {
-          current = current == null ? value : current + " " + value;
-          if (current.length() >= minTextLength) {
-            queue.add(current);
-            current = null;
-          }
-        }
-      }
-    }
-
-    public void setParseException(Exception ex) {
-      this.parseException = ex;
-    }
-
-    public synchronized boolean waitForNext(long pollTime) throws IOException {
-      if (!documentEnded) {
-        try {
-          wait(pollTime);
-        } catch (InterruptedException ex) {
-          // continue;
-        }
-      }
-      return documentEnded;
-    }
-
-    @Override
-    public synchronized void endDocument() throws SAXException {
-      this.documentEnded = true;
-      notify();
-    }
-
-    public String getCurrent() throws IOException {
-      checkParseException();
-      String value = queue.poll();
-      if (value == null && documentEnded) {
-        return current;
-      } else {
-        return value;
-      }
-    }
-    public void checkParseException() throws IOException {
-      if (parseException != null) {
-        throw new IOException(parseException);
-      }
-    }
-
-    public void setMinTextLength(int minTextLength) {
-      this.minTextLength = minTextLength;
-    }
+  static class ContentHandlerImpl extends ToTextContentHandler {
```

Comment:
This class doesn't do anything, just use ToTextContentHandler, or even better use Tika.toString()

Reply:
I kept a dedicated extension because I was imagining there could be a need to customize, stripping off or replacing some characters (say replacing all the insignificant whitespaces between with a single one, dropping the characters if the size of the buffer has reached some max amount, etc). But I'm OK with dropping a custom extension till we have the concrete ideas on what to customize...